2022Swerve/src/main/java/frc/robot/subsystems/SwerveSubsystem.java

```diff
@@ -0,0 +1,82 @@
+package frc.robot.subsystems;
+
+import com.kauailabs.navx.frc.AHRS;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.SwerveDriveKinematics;
+import edu.wpi.first.math.kinematics.SwerveModuleState;
+import edu.wpi.first.wpilibj.SPI;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.InstantCommand;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj2.command.WaitUntilCommand;
+import frc.robot.Constants.DriveConstants;
+import frc.robot.hardware.AbsoluteEncoders;
+import frc.robot.hardware.MotorControllers;
+
+public class SwerveSubsystem extends SubsystemBase{
+    private final SwerveModule frontLeft = new SwerveModule(
+        MotorControllers.FrontLeftModuleDrive.getMotor(),
+        MotorControllers.FrontLeftModuleTurn.getMotor(),
+        AbsoluteEncoders.FrontLeftModule.getSupplier());
+
+    private final SwerveModule frontRight = new SwerveModule(
+        MotorControllers.FrontRightModuleDrive.getMotor(),
+        MotorControllers.FrontRightModuleTurn.getMotor(),
+        AbsoluteEncoders.FrontRightModule.getSupplier());
+
+    private final SwerveModule backLeft = new SwerveModule(
+        MotorControllers.BackLeftModuleDrive.getMotor(),
+        MotorControllers.BackLeftModuleTurn.getMotor(),
+        AbsoluteEncoders.BackLeftModule.getSupplier());
+
+    private final SwerveModule backRight = new SwerveModule(
+        MotorControllers.BackRightModuleDrive.getMotor(),
+        MotorControllers.BackRightModuleTurn.getMotor(),
+        AbsoluteEncoders.BackRightModule.getSupplier());
+
+    private AHRS gyro = new AHRS(SPI.Port.kMXP);
+
+    public SwerveSubsystem() {
+        new WaitUntilCommand(this::gyroReady)
+        .andThen(new InstantCommand(this::zeroHeading,this))
+        .schedule();
```

Comment:
Is it necessary to call .schedule() here? Or is the command scheduled automatically?

Reply:
Yes, looking around, seems that you do need the schedule.

Many of the other examples are showing sending the command to a button mapping, or other mapping.